src/serial/serialcommands.cpp

```diff
@@ -35,16 +35,61 @@ namespace SerialCommands {
     CmdBuffer<64> cmdBuffer;
 
     void cmdSet(CmdParser * parser) {
-        if(parser->getParamCount() != 1 && parser->equalCmdParam(1, "WIFI")  ) {
-            if(parser->getParamCount() < 3) {
-                logger.error("CMD SET WIFI ERROR: Too few arguments");
-                logger.info("Syntax: SET WIFI \"<SSID>\" \"<PASSWORD>\"");
+        logger.error("CMD SET ERROR: Unrecognized variable to set");
+    }
+
+    void cmdWiFi(CmdParser* parser) {
+        if (parser->getParamCount() < 1) {
+            logger.error("CMD WIFI ERROR: No operation specified");
+            return;
+        }
+
+        if (parser->equalCmdParam(1, "ADD")) {
+            if (parser->getParamCount() < 3) {
+                logger.error("CMD WIFI ADD ERROR: Too few arguments");
+                logger.info("Syntax: WIFI ADD \"<SSID>\" \"<PASSWORD>\"");
             } else {
-                WiFiNetwork::setWiFiCredentials(parser->getCmdParam(2), parser->getCmdParam(3));
-                logger.info("CMD SET WIFI OK: New wifi credentials set, reconnecting");
+                const char* ssid = parser->getCmdParam(2);
+                const char* password = parser->getCmdParam(3);
+
+                if (strlen(ssid) >= 32) {
+                    logger.error("CMD WIFI ADD ERROR: SSID too long");
+                    return;
+                }
+
+                if (strlen(password) >= 64) {
+                    logger.error("CMD WIFI ADD ERROR: Password too long");
+                    return;
+                }
+
+                SlimeVR::Configuration::WiFiCredential wifiCredential;
+
+                strncpy(wifiCredential.ssid, ssid, 32);
```

Comment:
magic numbers, what do they mean

Reply:
anna already complained about too many comments in another PR :joy: